src/bentoml/_internal/cloud/deployment.py

```diff
@@ -73,26 +73,25 @@ class DeploymentConfigParameters:
     def verify(
         self,
     ):
-        deploy_by_param = (
-            self.name
-            or self.bento
-            or self.cluster
-            or self.access_authorization
-            or self.scaling_min
-            or self.scaling_max
-            or self.instance_type
-            or self.strategy
-            or self.envs
-            or self.extras
+        deploy_by_param = any(
+            param is not None and len(param) > 0
```

Comment:
I am afraid you can't judge with this, given there are integer values (scaling_min and scaling_max)
Can we just filter None, meaning it is specified by the users

Reply:
Yeah initially that was the case but it then the default for envs is a list so it will always evaluate to True if we dont check for it